src/index.js

```diff
@@ -61,83 +56,61 @@ class IpfsRepo {
    * Initialize a new repo.
    *
    * @param {Object} config - config to write into `config`.
-   * @param {function(Error)} callback
-   * @returns {void}
+   * @returns {Promise<void>}
    */
-  init (config, callback) {
+  async init (config) {
     log('initializing at: %s', this.path)
-
-    series([
-      (cb) => this.root.open(ignoringAlreadyOpened(cb)),
-      (cb) => this.config.set(buildConfig(config), cb),
-      (cb) => this.spec.set(buildDatastoreSpec(config), cb),
-      (cb) => this.version.set(repoVersion, cb)
-    ], callback)
+    await this.root.open()
+    await this.config.set(buildConfig(config))
+    await this.spec.set(buildDatastoreSpec(config))
+    await this.version.set(repoVersion)
   }
 
   /**
    * Open the repo. If the repo is already open no action will be taken.
    * If the repo is not initialized it will return an error.
```

Comment:
I believe that "If the repo is already open no action will be taken." is not really correct, because according to the L77, it will throw an error. Maybe lets update it?

Reply:
Good catch, the comment is definitely off 👍

Reply:
Maybe the error thrown should be bit more custom (eq. creating a new category in `./errors/index.js`) so people could detect if the opening of repo failed because it is locked and handle it? But not sure if it is in the scope of this PR though...